### PR TITLE
WP 685

### PR DESCRIPTION
**Pull request by dpegasusm**

### CODE REVIEW

File: inc/class-multitaxo-plugin.php [hunk 1]
```diff
@@ -257,13 +257,16 @@ public function multisite_set_screen_option( $status, $option, $value ) {
...
+
+		// Check that we have something.
+		if ( empty( $tax_slug ) ) {
+			wp_die( esc_html__( 'Invalid taxonomy.', 'multitaxo' ) );
 		}
 
-		$tax = get_multisite_taxonomy( $taxnow );
+		$tax = get_multisite_taxonomy( $tax_slug );
 
 		if ( ! $tax ) {
```
Comment (guillaumemolter) [inline inc/class-multitaxo-plugin.php]: We should use is_a($tax,'Multisite_Taxonomy'), also $mulsite_taxonomy seems like a more appropriate variable name.

File: inc/class-multitaxo-plugin.php [hunk 2]
```diff
@@ -677,6 +681,29 @@ public function ajax_inline_save_multisite_tag() {
...
+	 * @return void
+	 */
+	public function display_multisite_taxonomy_list() {
+		?>
+		<div class="wrap">
+		<h1><?php echo esc_html__( 'Multisite Taxonomies', 'multitaxo' ); ?></h1>
+		<ul>
+		<?php
+
+		$taxonomies = get_multisite_taxonomies( array(), 'objects' );
```
Comment (guillaumemolter) [inline inc/class-multitaxo-plugin.php:697]: What if there is no multisite taxonomies registered?

File: inc/class-multitaxo-plugin.php [hunk 3]
```diff
@@ -677,6 +681,29 @@ public function ajax_inline_save_multisite_tag() {
...
+
+		foreach ( $taxonomies as $tax_slug => $tax ) {
+			echo '<li><a href="admin.php?page=multisite_term_list_' . esc_attr( $tax_slug ) . '">' . esc_html( $tax->label ) . '</a></li>';
```
Comment (guillaumemolter) [inline inc/class-multitaxo-plugin.php]: ~~should we use esc_url?~~

**Review by guillaumemolter — CHANGES_REQUESTED** [review]

File: inc/class-multitaxo-plugin.php [hunk 4]
```diff
@@ -201,6 +201,7 @@ public function create_database_tables() {
 	 * @return void
 	 */
 	public function delete_database_tables() {
```
Comment (guillaumemolter) [inline inc/class-multitaxo-plugin.php:203]: Maybe we shoudl remove this empty function instead of adding comments to make it pass the tests ;-)

**Review by guillaumemolter — CHANGES_REQUESTED** [review]

**Comment by guillaumemolter** [comment]
Also do we need this screen? Can't wee default the top level menu to the first multisite taxonomy (which is the default behavior of most WP menus)?
screen shot 2018-03-23 at 11 54 20

File: inc/class-multitaxo-plugin.php (hunk 3 above)
Comment (guillaumemolter) [reply inc/class-multitaxo-plugin.php]: Sorry...wasn't thinking through it...this is good 👍

**Comment by dpegasusm** [comment]
I added that screen because in the future we talked about a screen for adding and removal of multisite taxonomies dynamically. I also think that having it go to the first is an unexpected user interaction. In all other cases on the menu the first section is something expected; Posts -> all posts screen, Media -> All media screen this is similar although far more sparse of an interaction.

**Review by guillaumemolter — APPROVED** [review]

**Pull request merged**